tests/testapps/AddressBookSample/AddressBook/SingleContactViewController.m

```diff
@@ -0,0 +1,417 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import "SingleContactViewController.h"
+#import "ViewController.h"
+#import <AddressBook/ABMutableMultiValue.h>
+#import <AddressBook/ABPerson.h>
+
+@interface SingleContactViewController ()
+
+/**
+ * Sets all of the text fields to have an 'enabled' state as given by
+ * enabled.
+ */
+- (void)_setTextFieldEnabledState:(BOOL)enabled;
+
+/**
+ * Populates all of the text fields (name, phone number, etc) with
+ * the information obtained from self.contact.
+ */
+- (void)_populateTextFields;
+
+/**
+ * Populates the first name text field with the first name of the
+ * contact in self.contact.
+ */
+- (void)_populateFirstName;
+
+/**
+ * Populates the last name text field with the last name of the
+ * contact in self.contact.
+ */
+- (void)_populateLastName;
+
+/**
+ * Populates the phone number text field with the first phone number
+ * listed in the phone numbers of the contact in self.contact.
+ */
+- (void)_populatePhoneNumber;
+
+/**
+ * Populates the email text field with the first email address
+ * listed in the emails of the contact in self.contact.
+ */
+- (void)_populateEmail;
+
+/**
+ * Populates the address text field with the street of the first address
+ * listed in the addresses of the contact in self.contact.
+ */
+- (void)_populateAddress;
+
+/**
+ * Populates the contact photo to use the contact image (if present) of
+ * the contact in self.contact.
+ */
+- (void)_populateContactPhoto;
+
+/**
+ * Updates the contact in self.contact to now reflect the values in the
+ * text fields. If a text field is blank, that property is removed from the
+ * contact. The user can only modify values that previously had a value --
+ * they cannot add additional ones (they can remove existing values, though).
+ */
+- (void)_modifyContact;
+
+/**
+ * Creates a new contact using the values in the text fields, and then adds
+ * said contact to the AddressBook. The phone number is assigned with the label of
+ * Main Phone, and the email is assigned with a label of Work. The address creates
+ * an address with the given value as the street, with a label of Home.
+ */
+- (void)_createContact;
+
+/**
+ * Returns the value in the text field of the given UITextField*, or
+ * NULL if the text field has no text in it.
+ */
+- (CFTypeRef)_copyContactInfoFrom:(UITextField*)textField;
+
+@end
+
+@implementation SingleContactViewController
+
+- (void)dealloc {
+    // Viewing, modifying, and deleting a contact uses the parent's ABAddressBookRef.
+    // Create, however, has its own, and so it must clean up when finished.
+    if (self.contactOperation == kAddressBookCreateContact) {
+        CFRelease(self.addressBook);
+    }
```

Comment:
ARC prohibits explicit calls to dealloc -- it automatically does the [super dealloc].

---
In reply to: [70119973](https://github.com/Microsoft/WinObjC/pull/627#discussion_r70119973) [](ancestors = 70119973)